### PR TITLE
Sscs 11348 ut decision refused

### PR DESCRIPTION
**Pull request by paul-pearson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-11348

### Change description ###

ut decision refused

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    private final boolean isPostHearingsBEnabled;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.SUBMITTED)
+                && callback.getEvent() == EventType.SEND_TO_FIRST_TIER
+                && isPostHearingsEnabled && isPostHearingsBEnabled;
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java:37]: you can shorten it and just check for the post hearings b flag

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+        caseData = ccdCallbackMapService.handleCcdCallbackMap(caseData.getPostHearing().getSendToFirstTier().getAction(),
+                callback.getCaseDetails().getCaseData());
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java:45]: use "caseData", youve made the object on line 43

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    @BeforeEach
+    void setUp() {
+        handler = new SendToFirstTierSubmittedHandler(ccdCallbackMapService, true, true);
+
+        caseData = SscsCaseData.builder()
+                .ccdCaseId(String.valueOf(CASE_ID))
+                .postHearing(PostHearing.builder()
+                        .sendToFirstTier(SendToFirstTier.builder().build())
+                        .build())
```
Comment (nickhill111) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java:51]: check the tabbing in this file, seems like its twice what its meant to be

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Test
+    void givenAValidSubmittedEvent_thenReturnTrue() {
+        when(callback.getEvent()).thenReturn(SEND_TO_FIRST_TIER);
+        assertThat(handler.canHandle(SUBMITTED, callback)).isTrue();
+    }
+
+    @Test
+    void givenAInvalidEvent_thenReturnFalse() {
```
Comment (nickhill111) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java:62]: missing some CC: https://sonarcloud.io/component_measures?metric=new_coverage&selected=SSCSTCA%3Asrc%2Fmain%2Fjava%2Fuk%2Fgov%2Fhmcts%2Freform%2Fsscs%2Fccd%2Fpresubmit%2Fsendtofirsttier%2FSendToFirstTierSubmittedHandler.java&view=list&pullRequest=3380&id=SSCSTCA

**Review by nickhill111 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java (hunk 1 above)
Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java]: Done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java (hunk 2 above)
Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java]: Done

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java (hunk 3 above)
Comment (paul-pearson) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java]: Done

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java (hunk 4 above)
Comment (paul-pearson) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java:62]: Test added

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        assertThat(handler.canHandle(SUBMITTED, callback)).isFalse();
+    }
+
+    @Test
+    void givenAInvalidCallbackType_thenReturnFalse() {
+        assertThat(handler.canHandle(MID_EVENT, callback)).isFalse();
+    }
+
+    @Test
+    void givenPostHearingsEnabledFalse_thenReturnFalse() {
```
Comment (nickhill111) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java:80]: need to rename this test

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.SUBMITTED)
+                && callback.getEvent() == EventType.SEND_TO_FIRST_TIER
+                && isPostHearingsBEnabled;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java:39]: need to call canHandle() at the start of the method, check other handlers to see the generic way of doing it

**Review by nickhill111 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java (hunk 5 above)
Comment (paul-pearson) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandlerTest.java]: Test is a duplicate - deleted.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java (hunk 6 above)
Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java:39]: Done

**Review by nickhill111 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierAboutToSubmitHandler.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        if (sendToFirstTier.getAction() == null) {
+            response.addError("There is no decision type");
+        }
+        if (sendToFirstTier.getDecisionDocument() == null) {
+            response.addError("There is no decision document");
+        }
+        return response;
+    }
+
+    private DocumentType getSendToFirstTierDocumentType(SendToFirstTierActions action) {
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierAboutToSubmitHandler.java:73]: might want to check if the action is null or add the @nonNull annotation to the parameter

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+        caseData = ccdCallbackMapService.handleCcdCallbackMap(caseData.getPostHearing().getSendToFirstTier().getAction(), caseData);
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierSubmittedHandler.java:44]: space this out a bit

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierAboutToStartHandler.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierAboutToStartHandler.java:42]: space this out a bit

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierAboutToSubmitHandler.java (hunk 7 above)
Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/sendtofirsttier/SendToFirstTierAboutToSubmitHandler.java]: Done

**Review by nickhill111 — APPROVED** [review]

**Review by DGordonV1 — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]

**Review by AndreiTrushkov — APPROVED** [review]

**Review by DGordonV1 — DISMISSED** [review]

**Review by nickhill111 — APPROVED** [review]

**Review by DGordonV1 — APPROVED** [review]

**Pull request merged**